Examples/Swift/AppDelegate.swift

```diff
@@ -72,7 +72,7 @@ extension AppDelegate: CarPlayManagerDelegate {
             
             if !navigationViewControllerExistsInStack {
                 
-                let locationManager = routeController.locationManager
+                let locationManager = routeController.locationManager.copy() as! NavigationLocationManager
```

Comment:
can this fail?

Reply:
It can fail but it won't crash, e.g. using a `ReplayLocationManager`, you'd get something that resembles a `NavigationLocationManager` w/o the state of the replaying location manager. The superclass `(NavigationLocationManager)` is enforced by the route controller. So this could be kinda volatile in some scenarios.

I wonder if another pattern could help us here, e.g. a delegate with multiple receivers since the GPS is shared between CarPlay and device.

Something like:
```swift
var receivers: [CLLocationManagerDelegate]?
func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
    receivers?.forEach { $0.locationManager(manager, didUpdateLocations: locations) }
}
```
which is something usually more seen in Java. 🤔

Reply:
yeah that's basically what I meant by message passing

Reply:
A delegator can only have one delegate; otherwise, it’s an observer pattern rather than the delegation pattern, so we’d use NSNotificationCenter.

Reply:
Changing this to a loosely coupled observer pattern such as NSNotificationCenter would be a pretty big breaking change and could lead to many unanticipated impacts. Especially when the nav service branch gets merged which uses more than one location manager simultaneously.